src/main/java/aztech/modern_industrialization/machines/MachineBlockEntity.java

```diff
@@ -75,8 +71,8 @@
 @SuppressWarnings("rawtypes")
 public abstract class MachineBlockEntity extends FastBlockEntity
         implements MenuProvider, WrenchableBlockEntity {
-    public final List<GuiComponent.Server> guiComponents = new ArrayList<>();
-    private final List<IComponent> icomponents = new ArrayList<>();
+    protected final ComponentStorage.GuiServer guiComponents = new ComponentStorage.GuiServer();
+    protected final ComponentStorage.Server icomponents = new ComponentStorage.Server();
```

Comment:
I think this can be renamed to `components`.

Reply:
Agreed. That was something I wanted to question but had forgotten